From: Juan Navarro <juan.navarro@gmx.es>
Date: Thu, 2 Aug 2018 15:50:09 +0200
Subject: Kurento: Add build config for GStreamer 1.5

---
 configure.ac    | 27 +++++++++++++++++++++++++++
 gst/Makefile.am | 17 +++++++++++++++++
 2 files changed, 44 insertions(+)

diff --git a/configure.ac b/configure.ac
index 30fd7ee..cd2c201 100644
--- a/configure.ac
+++ b/configure.ac
@@ -244,6 +244,11 @@ AC_ARG_WITH(gstreamer,
 	[with_gstreamer=${withval}],
 	[with_gstreamer=auto])
 
+AC_ARG_WITH(gstreamer-1.5,
+	AS_HELP_STRING([--with-gstreamer-1.5],[build GStreamer 1.5 plugin]),
+	[with_gstreamer15=${withval}],
+	[with_gstreamer15=auto])
+
 AC_ARG_WITH(gstreamer-0.10, 
 	AS_HELP_STRING([--with-gstreamer-0.10],[build GStreamer 0.10 plugin]),
 	[with_gstreamer010=${withval}],
@@ -279,6 +284,26 @@ AS_IF([test "$with_gstreamer" != no], [
 		])
 ])
 
+AS_IF([test "$with_gstreamer15" != no], [
+
+	PKG_CHECK_MODULES(GST15, [
+		gstreamer-1.5 >= 1.5
+		gstreamer-base-1.5 >= 1.5
+		],
+		[
+		with_gstreamer15=yes
+		GST_MAJORMINOR=1.5
+		gstplugin15dir="\$(libdir)/gstreamer-$GST_MAJORMINOR"
+		],
+		[
+		AS_IF([test "$with_gstreamer" = yes], [
+			AC_MSG_ERROR([GStreamer 1.5 support was requested but GStreamer 1.5 libraries are not available])
+		])
+
+		with_gstreamer15=no
+		])
+])
+
 AS_IF([test "$with_gstreamer010" != no], [
   
 	PKG_CHECK_MODULES(GST010, [
@@ -300,9 +325,11 @@ AS_IF([test "$with_gstreamer010" != no], [
 ])
 
 AC_SUBST(gstplugindir)
+AC_SUBST(gstplugin15dir)
 AC_SUBST(gstplugin010dir)
 
 AM_CONDITIONAL(WITH_GSTREAMER, test "$with_gstreamer" = yes)
+AM_CONDITIONAL(WITH_GSTREAMER15, test "$with_gstreamer15" = yes)
 AM_CONDITIONAL(HAVE_GST_CHECK, test "$have_gst_check" = yes)
 AM_CONDITIONAL(WITH_GSTREAMER010, test "$with_gstreamer010" = yes)
 
diff --git a/gst/Makefile.am b/gst/Makefile.am
index 8776780..3f771c9 100644
--- a/gst/Makefile.am
+++ b/gst/Makefile.am
@@ -36,6 +36,23 @@ libgstnice_la_LIBADD = 	$(COMMON_LIBADD) $(GST_LIBS)
 libgstnice_la_LDFLAGS = -module -avoid-version -no-undefined
 endif
 
+if WITH_GSTREAMER15
+gstplugin15_LTLIBRARIES = libgstnice15.la
+
+libgstnice15_la_CFLAGS = $(AM_CFLAGS) $(GST15_CFLAGS) -DGST_USE_UNSTABLE_API
+
+libgstnice15_la_SOURCES = \
+	gstnicesrc.h \
+	gstnicesrc.c \
+	gstnicesink.h \
+	gstnicesink.c \
+	gstnice.h \
+	gstnice.c
+
+libgstnice15_la_LIBADD = $(COMMON_LIBADD) $(GST15_LIBS)
+
+libgstnice15_la_LDFLAGS = -module -avoid-version
+endif
 
 if WITH_GSTREAMER010
 gstplugin010_LTLIBRARIES = libgstnice010.la
